### PR TITLE
403-fix : React  key error  is fixed

### LINKED ISSUE #403: React error in JavaScript Pre School RU course page
## 🐛 Describe the Bug

Image

## 🔄 Steps to Reproduce

Navigate to JavaScript Pre School RU course page

### PR DESCRIPTION
**Pull request by ismoilkhon**
## What type of PR is this? (select all that apply)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description
React error in JavaScript Pre School RU course page is fixsed

## Related Tickets & Documents

- Related Issue #403 
- Closes #403 

## Screenshots, Recordings
Before
issu-434
After
Screen Shot 2024-07-11 at 18 04 52

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

## Summary by CodeRabbit

- **Refactor**
  - Improved internal logic for mapping over items in the `Required` and `Trainers` components.
  - Enhanced rendering performance by optimizing key assignments in lists.

### CODE REVIEW

**Review by Quiddlee — APPROVED** [review]

**Comment by Quiddlee** [comment]
Please change related tickets section according to this:

image

**Review by Wystov — APPROVED** [review]
Actually using index as key is not a best practice, but for not dynamic content it should work fine.

File: src/widgets/required/ui/required.tsx [hunk 1]
```diff
@@ -40,9 +42,9 @@ export const Required = ({ courseName, marked1, marked2 }: RequiredProps) => {
           )}
           <div className="will-learn">
             {isWillLearnExist
-              ? willLearn.map((willLearn) => {
+              ? willLearn.map((willLearn, index) => {
```
Comment (natanchik) [inline src/widgets/required/ui/required.tsx:45]: I suggest:
`{isWillLearnExist && willLearn.map((willLearn, index) => {
                return (
                  <article key={index}>
                    <Subtitle text={willLearn.title} />
                    <Actions actions={willLearn.description} marked={marked2} />
                  </article>
                );
              })}`

Comment (ismoilkhon) [reply src/widgets/required/ui/required.tsx:45]: done

**Pull request merged**